### ISSUE TITLE
introspection: Hint returned when auto running on mz_introspection could be improved

### ISSUE THREAD

**Issue by ParkMyCar** [MaterializeInc/materialize#21307]
### What version of Materialize are you using?

v0.66.0 (647282c5d)

### What is the issue?

We received some feedback that the hint returned when a query gets auto run on the mz_introspection cluster could be improved.

>Super-early doors: This message in the shell  confused me
> `DEBUG: query was automatically run on the "mz_introspection" cluster`
> Is it bad? Did I do something wrong? Googling the message brought me to [this GH issue](https://github.com/MaterializeInc/materialize/issues/19413), which explains nicely. But maybe there's a little more info we can add to the message?

Adding a HINT to the returned message which explains it in more detail, or extending the error message to be something like: "query was automatically run on the 'mz_introspection' cluster **for improved performance**" could be good.

**Comment by benesch** [comment]
> This hint is already `debug` level. Why are we showing debug level hints in the SQL shell by default?

Indeed! That's a separate issue and one we're tracking here: https://github.com/MaterializeInc/console/issues/681.

**Comment by benesch** [comment]
Hot take: having this message at `DEBUG` level is fairly pointless. Approximately no one will ever see it, once we fix https://github.com/MaterializeInc/console/issues/681. The number of people who set `client_min_messages` to `debug` probably rounds to zero.

I propose that we:

  * Make this message more useful and less scary by adjusting it to something like
    ```
    NOTICE: query automatically routed to the "mz_introspection" cluster to improve performance
    HINT: See https://materialize.com/s/auto-routing for details.
    ```
  * Upgrade the message from `debug` to `notice` so that it is present by default.

This adds a touch more information to the notice ("to improve performance"), as suggested by @ParkMyCar, that I think should satisfy most users' curiosity. I don't think we want to go into much more detail, because we'll be sending this notice for _every_ such query, and it'll be fairly spammy. But the hint can link to docs that expound upon auto routing, including details about how to disable it on the off chance that you want to force one of these queries to run on the active cluster.
